src/clusterfuzz/_internal/bot/tasks/utasks/corpus_pruning_task.py

```diff
@@ -548,12 +549,31 @@ def run(self, timeout):
     return result.stats
 
 
-def _record_cross_pollination_stats(stats):
-  """Log stats about cross pollination in BigQuery."""
-  # If no stats were gathered due to a timeout or lack of corpus, return.
+def _fill_cross_pollination_stats(stats, output):
```

Comment:
nit: is it possible to be consistent with `_extract_coverage_information` below and return the proto object here instead of using `CopyFrom`?

Reply:
The issue with this is the possibility of the `cross_pollination_stats` being None. 
We cannot set a proto field to None (even in the constructor unfortunately).
If I set them to an empty field, I wouldn't be able to check if the field was set from postprocess unfortunately

Reply:
A workround would be to add a bool field in `CrossPollinationStatistics` and use it to tell if the field is empty and is to be ignored for example.

Reply:
I see, interesting!